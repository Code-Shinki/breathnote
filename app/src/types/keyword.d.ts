export type AllKeywords = {
  contents: Keyword[]
  totalCount: number
  offset: number
  limit: number
}

export type Keyword = {
  id: string
  createdAt: string
  updatedAt: string
  publishedAt: string
  revisedAt: string
  name: string
  slug: string
  description: string
}
