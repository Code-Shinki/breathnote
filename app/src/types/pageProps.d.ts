import { Keyword } from 'types/keyword'

export type PageProps = {
  url: string
  type: string
  title: string
  description: string
  thumbnail?: string
  noindex?: boolean
  keywords: Keyword[]
}
