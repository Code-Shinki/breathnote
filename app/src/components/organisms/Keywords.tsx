import Link from 'next/link'
import React from 'react'
import { Keyword } from 'types/keyword'
import styles from './Keywords.module.scss'

type Props = {
  keywords: Keyword[]
}

const Keywords: React.FC<Props> = ({ keywords }) => {
  const items = keywords.map((keyword) => (
    <li key={keyword.slug}>
      <Link href={`/keywords/${keyword.slug}`} prefetch={false}>
        <a>
          <span>{keyword.name}</span>
        </a>
      </Link>
    </li>
  ))

  return (
    <section className={styles.section}>
      <h2 className={styles.title}>Keywords</h2>
      <ul className={styles.list}>{items}</ul>
    </section>
  )
}

export default Keywords
