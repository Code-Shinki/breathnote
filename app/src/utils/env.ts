export const SITE_DOMAIN = process.env.NEXT_PUBLIC_SITE_DOMAIN || ''
export const SITE_TITLE = process.env.NEXT_PUBLIC_SITE_TITLE || ''
export const SITE_SUBTITLE = process.env.NEXT_PUBLIC_SITE_SUBTITLE || ''
export const SITE_DESCRIPTION = process.env.NEXT_PUBLIC_SITE_DESCRIPTION || ''
export const ARTICLE_PER_PAGE = Number(process.env.NEXT_PUBLIC_ARTICLE_PER_PAGE) || 12
export const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING_ID || ''
export const REVALIDATE_TIME = Number(process.env.REVALIDATE_TIME) || 60

export const API_KEY = process.env.API_KEY || ''
export const API_ENDPOINT = process.env.API_ENDPOINT || ''
export const DRAFT_TOKEN = process.env.DRAFT_TOKEN || ''
