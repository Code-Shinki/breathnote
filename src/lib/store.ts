/* ＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/

  LIB -> STORE

  - microCMS から非同期処理で取得したデータを保存

  - CONFIG : サイト設定 (オブジェクト)
  - ALL_POSTS : 記事一覧 (配列)
  - ALL_CATEGORIES : カテゴリ一覧 (配列)
  - ALL_TAGS : タグ一覧 (配列)

＿/ ＿/ ＿/ ＿/ ＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/ */

import { AllCategoriesType, AllConfigType, AllPostsType, AllTagsType } from 'lib/types'

export const CONFIG = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllConfigType = await fetch('https://breathnote.microcms.io/api/v1/settings/config', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData
})()

export const ALL_POSTS = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllPostsType = await fetch('https://breathnote.microcms.io/api/v1/posts', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()

export const ALL_CATEGORIES = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllCategoriesType = await fetch('https://breathnote.microcms.io/api/v1/categories', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()

export const ALL_TAGS = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllTagsType = await fetch('https://breathnote.microcms.io/api/v1/tags', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()
