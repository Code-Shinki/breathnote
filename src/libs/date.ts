import { format } from 'date-fns'
import { utcToZonedTime } from 'date-fns-tz'

export const getFormattedDate = (date: string | Date | null, conversion: string) => {
  // 下書き中は任意の文字列を返す
  if (!date) return 'Draft'

  const jstDate = utcToZonedTime(new Date(date), 'Asia/Tokyo')

  return format(jstDate, conversion)
}
