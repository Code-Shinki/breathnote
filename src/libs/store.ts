import axios from 'axios'
import { AllKeywords } from 'types/keyword'
import { AllPosts } from 'types/post'
import { API_ENDPOINT, API_KEY } from 'utils/env'

const KEY = { headers: { 'X-API-KEY': API_KEY } }

const fetchAllPosts = async () => {
  console.log('ALL_POSTS fire')

  // 一度に受け取る記事数
  const limit = 10

  // 総記事数等を含む最初の10記事を取得
  const posts: AllPosts = await axios
    .get(`${API_ENDPOINT}/posts?limit=${limit}`, KEY)
    .then((res) => res.data)
    .catch((err) => {
      throw new Error(`FETCH FAILED (fetchAllPosts() - First reqest) : ${err}`)
    })

  // APIを叩く回数
  const requestsCount = Math.floor(posts.totalCount / limit)

  // すべての記事を取得してposts.contentsに追加
  for (let i = 1; i <= requestsCount; i++) {
    const fetchData = await axios
      .get(`${API_ENDPOINT}/posts?offset=${limit * i}&limit=${limit}`, KEY)
      .then((res) => res.data)
      .catch((err) => {
        throw new Error(`FETCH FAILED (fetchAllPosts() - Roop ${i} reqest) : ${err}`)
      })
    posts.contents = posts.contents.concat(fetchData.contents)
  }

  // 記事を降順 (公開日が新しい順) に並び替え
  if (posts.contents.length > 1) {
    posts.contents.sort((a, b) => {
      if (a.publishedAt < b.publishedAt) {
        return 1
      } else {
        return -1
      }
    })
  }

  return posts
}

const fetchAllKeywords = async () => {
  console.log('ALL_KEYWORDS fire')

  // 一度に受け取るキーワード数
  const limit = 10

  // 総キーワード数等を含む最初の10データを取得
  const keywords: AllKeywords = await axios
    .get(`${API_ENDPOINT}/keywords?limit=${limit}`, KEY)
    .then((res) => res.data)
    .catch((err) => {
      throw new Error(`FETCH FAILED (fetchAllKeywords() - First reqest) : ${err}`)
    })

  // APIを叩く回数
  const requestsCount = Math.floor(keywords.totalCount / limit)

  // すべてのキーワードを取得してposts.contentsに追加
  for (let i = 1; i <= requestsCount; i++) {
    const fetchData = await axios
      .get(`${API_ENDPOINT}/keywords?offset=${limit * i}&limit=${limit}`, KEY)
      .then((res) => res.data)
      .catch((err) => {
        throw new Error(`FETCH FAILED (fetchAllKeywords() - Roop ${i} reqest) : ${err}`)
      })
    keywords.contents = keywords.contents.concat(fetchData.contents)
  }

  // キーワードを昇順 (アルファベット順) に並び替え
  if (keywords.contents.length > 1) {
    keywords.contents.sort((a, b) => {
      if (a.name > b.name) {
        return 1
      } else {
        return -1
      }
    })
  }

  return keywords
}

export const ALL_POSTS = fetchAllPosts()
export const ALL_KEYWORDS = fetchAllKeywords()
