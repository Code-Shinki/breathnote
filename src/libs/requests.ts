import { ALL_POSTS } from 'libs/store'

// keywords/index.tsx, keywords/page/[offset].tsx
export const getKeywordPosts = async (slug: string) => {
  const posts = (await ALL_POSTS).contents
  const keywordPosts = posts.filter((post) => {
    return post.keywords.some((keyword) => {
      return keyword.slug === slug
    })
  })
  return keywordPosts
}
