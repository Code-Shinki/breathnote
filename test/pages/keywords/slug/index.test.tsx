import { cleanup, render } from '@testing-library/react'
import React from 'react'
import { getKeywordPosts } from '../../../../src/libs/requests'
import { fetchAllKeywords } from '../../../../src/libs/store'
import Keywords from '../../../../src/pages/keywords/[slug]/index'
import { Keyword } from '../../../../src/types/keyword'
import { SITE_DOMAIN, SITE_TITLE } from '../../../../src/utils/env'

describe(`KEYWORD ARTICLE PAGE 1/7 (pages/keywords/[slug]/index.tsx)`, () => {
  afterEach(cleanup)

  test('snapshot', async () => {
    const keywords = (await fetchAllKeywords()).contents
    const keyword = keywords.find((keyword) => keyword.slug === 'nextjs') as Keyword
    const posts = (await getKeywordPosts(keyword.slug)).slice(-1)
    const postLength = 84
    const pageProps = {
      url: `${SITE_DOMAIN}/keywords/${keyword.slug}`,
      type: 'keywords',
      title: `${keyword.slug} - ${SITE_TITLE}`,
      description: keyword.slug,
      noindex: true,
      keywords: keywords,
    }

    const { asFragment } = render(
      <Keywords pageProps={pageProps} posts={posts} postLength={postLength} keyword={keyword} />
    )
    const tree = asFragment()
    expect(tree).toMatchSnapshot()
  })
})
