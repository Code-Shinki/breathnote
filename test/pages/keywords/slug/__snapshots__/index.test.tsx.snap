// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KEYWORD ARTICLE PAGE 1/7 (pages/keywords/[slug]/index.tsx) snapshot 1`] = `
<DocumentFragment>
  <header
    class="max-w-5xl px-3 sm:px-5 py-6 md:py-9 lg:py-12 mx-auto flex justify-between items-center"
  >
    <div
      class="text-2xl sm:text-3xl textShadow wave"
    >
      <a
        href="/"
      >
        <span
          class="text-accent-light dark:text-accent-dark"
        >
          B
        </span>
        reath Note
      </a>
    </div>
    <div
      class="flex"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <div
        class="relative inline-block w-12 mx-2 align-middle select-none"
      >
        <input
          class="toggle-checkbox"
          id="toggle"
          name="toggle"
          type="checkbox"
        />
        <label
          class="h-6 text-snow-200 dark:text-night-400 bg-snow-200 dark:bg-night-300 rounded-full overflow-hidden block cursor-pointer"
          for="toggle"
        >
          .
        </label>
      </div>
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </header>
  <main
    class="max-w-5xl px-3 sm:px-5 mx-auto"
  >
    <section>
      <h1
        class="mb-6 sm:mb-8 md:mb-10 text-xl sm:text-2xl md:text-3xl text-center"
      >
        <span
          class="text-2xl sm:text-3xl md:text-4xl text-accent-light dark:text-accent-dark tracking-wider"
        >
          Next.js
        </span>
        の記事
      </h1>
      <article
        class="p-3 md:p-4 lg:p-5 mb-6 md:mb-7 lg:mb-8 xl:mb-9 border border-shadow-light dark:border-shadow-dark rounded-lg shadow-light hover:shadow-none dark:shadow-dark dark:hover:shadow-none relative"
      >
        <h2
          class="mb-3 text-base sm:text-lg lg:text-xl"
        >
          ヘッドレスCMSから受け取ったマークダウン形式の記事にnext/imageを利用する
        </h2>
        <ul
          class="mb-4 text-xs sm:text-sm flex flex-wrap"
        >
          <li
            class="px-2 md:px-3 py-1 mr-2 sm:mr-3 md:mr-4 text-accent-light dark:text-accent-dark rounded border border-shadow-light dark:border-shadow-dark shadow-inset-light dark:shadow-inset-dark"
          >
            Next.js
          </li>
          <li
            class="px-2 md:px-3 py-1 mr-2 sm:mr-3 md:mr-4 text-accent-light dark:text-accent-dark rounded border border-shadow-light dark:border-shadow-dark shadow-inset-light dark:shadow-inset-dark"
          >
            Jamstack
          </li>
        </ul>
        <p
          class="mb-4 text-xs sm:text-sm leading-normal md:hidden"
        >
          ヘッドレスCMSのAPIから文字列として受け取ったマークダウンを、next-mdx-remoteというライブラリを用いてNext.jsコンポーネントに ...
        </p>
        <p
          class="mb-4 text-sm leading-normal hidden md:block"
        >
          ヘッドレスCMSのAPIから文字列として受け取ったマークダウンを、next-mdx-remoteというライブラリを用いてNext.jsコンポーネントに変換する方法を紹介しています。
        </p>
        <div
          class="flex justify-end items-center"
        >
          <svg
            class="w-4 md:w-5 h-4 md:h-5 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
          <time
            class="text-xs md:text-sm text-right block"
            datetime="2021-02-27T06:00:02.752Z"
            itemprop="datepublished"
          >
            2021年02月27日
          </time>
        </div>
        <a
          class="w-full h-full opacity-0 absolute top-0 left-0"
          href="/posts/nextjs-markdown-to-jsx"
        >
          Link
        </a>
      </article>
      <article
        class="p-3 md:p-4 lg:p-5 mb-6 md:mb-7 lg:mb-8 xl:mb-9 border border-shadow-light dark:border-shadow-dark rounded-lg shadow-light hover:shadow-none dark:shadow-dark dark:hover:shadow-none relative"
      >
        <h2
          class="mb-3 text-base sm:text-lg lg:text-xl"
        >
          JamstackなNext.jsブログにプレビューモードを実装する
        </h2>
        <ul
          class="mb-4 text-xs sm:text-sm flex flex-wrap"
        >
          <li
            class="px-2 md:px-3 py-1 mr-2 sm:mr-3 md:mr-4 text-accent-light dark:text-accent-dark rounded border border-shadow-light dark:border-shadow-dark shadow-inset-light dark:shadow-inset-dark"
          >
            Next.js
          </li>
          <li
            class="px-2 md:px-3 py-1 mr-2 sm:mr-3 md:mr-4 text-accent-light dark:text-accent-dark rounded border border-shadow-light dark:border-shadow-dark shadow-inset-light dark:shadow-inset-dark"
          >
            Jamstack
          </li>
        </ul>
        <p
          class="mb-4 text-xs sm:text-sm leading-normal md:hidden"
        >
          VercelへデプロイしているNext.js製のブログにプレビュー機能を追加します。本記事では、ヘッドレスCMSサービスのmicroCMSと連動したプ ...
        </p>
        <p
          class="mb-4 text-sm leading-normal hidden md:block"
        >
          VercelへデプロイしているNext.js製のブログにプレビュー機能を追加します。本記事では、ヘッドレスCMSサービスのmicroCMSと連動したプレビュー機能を実装しながら、セキュリティの設定やCookieの管理方法を説明しています。
        </p>
        <div
          class="flex justify-end items-center"
        >
          <svg
            class="w-4 md:w-5 h-4 md:h-5 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
          <time
            class="text-xs md:text-sm text-right block"
            datetime="2021-02-23T14:16:00.797Z"
            itemprop="datepublished"
          >
            2021年02月23日
          </time>
        </div>
        <a
          class="w-full h-full opacity-0 absolute top-0 left-0"
          href="/posts/nextjs-preview-mode"
        >
          Link
        </a>
      </article>
    </section>
    <nav
      aria-label="pagination"
      class="my-6"
    >
      <ul
        class="flex justify-center"
      >
        <li
          class="mx-1 md:mx-2 border border-shadow-light dark:border-shadow-dark rounded-full shadow-inset-light dark:shadow-inset-dark hover:border-shadow-light dark:hover:border-shadow-dark hover:shadow-inset-light dark:hover:shadow-inset-dark"
        >
          <a
            class="px-3 md:px-4 py-1 md:py-2 rounded-full block"
            href="/keywords/nextjs"
          >
            1
          </a>
        </li>
        <li
          class="mx-1 md:mx-2 border border-shadow-light dark:border-shadow-dark rounded-full nullhover:border-shadow-light dark:hover:border-shadow-dark hover:shadow-inset-light dark:hover:shadow-inset-dark"
        >
          <a
            class="px-3 md:px-4 py-1 md:py-2 rounded-full block"
            href="/keywords/nextjs/page/2"
          >
            2
          </a>
        </li>
        <li
          class="py-1 md:py-2 leading-none"
        >
          ...
        </li>
        <li
          class="mx-1 md:mx-2 border border-shadow-light dark:border-shadow-dark rounded-full nullhover:border-shadow-light dark:hover:border-shadow-dark hover:shadow-inset-light dark:hover:shadow-inset-dark"
        >
          <a
            class="px-3 md:px-4 py-1 md:py-2 rounded-full block"
            href="/keywords/nextjs/page/7"
          >
            7
          </a>
        </li>
      </ul>
    </nav>
    <aside
      class="mt-8 sm:mt-10 lg:mt-12"
    >
      <h2
        class="mb-3 sm:mb-5 lg:mb-6 text-xl sm:text-2xl tracking-wider flex justify-center items-center"
      >
        <span
          class="text-accent-light dark:text-accent-dark text-2xl sm:text-3xl"
        >
          S
        </span>
        HARE
      </h2>
      <div
        class="flex justify-center items-center"
      >
        <a
          class="w-11 sm:w-14 h-11 sm:h-14 mx-2 sm:mx-4 rounded-full flex justify-center items-center hover:text-sns-twitter border border-shadow-light dark:border-shadow-dark hover:border-sns-twitter dark:hover:border-sns-twitter shadow-light hover:shadow-none dark:shadow-inset-dark dark:hover:shadow-none"
          href="https://twitter.com/share?url=https%3A%2F%2Fblog.shinki.net%2Fkeywords%2Fnextjs"
          rel="nofollow noopener noreferrer"
          target="_blank"
          title="Twitterでシェアする"
        >
          <svg
            class="w-6 sm:w-7 h-6 sm:h-7"
            fill="currentColor"
            height="24"
            stroke="none"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 4.557c-.883.392-1.832.656-2.828.775 1.017-.609 1.798-1.574 2.165-2.724-.951.564-2.005.974-3.127 1.195-.897-.957-2.178-1.555-3.594-1.555-3.179 0-5.515 2.966-4.797 6.045-4.091-.205-7.719-2.165-10.148-5.144-1.29 2.213-.669 5.108 1.523 6.574-.806-.026-1.566-.247-2.229-.616-.054 2.281 1.581 4.415 3.949 4.89-.693.188-1.452.232-2.224.084.626 1.956 2.444 3.379 4.6 3.419-2.07 1.623-4.678 2.348-7.29 2.04 2.179 1.397 4.768 2.212 7.548 2.212 9.142 0 14.307-7.721 13.995-14.646.962-.695 1.797-1.562 2.457-2.549z"
            />
          </svg>
        </a>
        <a
          class="w-11 sm:w-14 h-11 sm:h-14 mx-2 sm:mx-4 rounded-full flex justify-center items-center hover:text-sns-facebook border border-shadow-light dark:border-shadow-dark hover:border-sns-facebook dark:hover:border-sns-facebook shadow-light hover:shadow-none dark:shadow-inset-dark dark:hover:shadow-none"
          href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fblog.shinki.net%2Fkeywords%2Fnextjs"
          rel="nofollow noopener noreferrer"
          target="_blank"
          title="Facebookでシェアする"
        >
          <svg
            class="w-6 sm:w-7 h-6 sm:h-7"
            fill="currentColor"
            height="24"
            stroke="none"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 8h-3v4h3v12h5v-12h3.642l.358-4h-4v-1.667c0-.955.192-1.333 1.115-1.333h2.885v-5h-3.808c-3.596 0-5.192 1.583-5.192 4.615v3.385z"
            />
          </svg>
        </a>
        <a
          class="w-11 sm:w-14 h-11 sm:h-14 mx-2 sm:mx-4 rounded-full flex justify-center items-center hover:text-sns-line border border-shadow-light dark:border-shadow-dark hover:border-sns-line dark:hover:border-sns-line shadow-light hover:shadow-none dark:shadow-inset-dark dark:hover:shadow-none"
          href="https://social-plugins.line.me/lineit/share?url=https%3A%2F%2Fblog.shinki.net%2Fkeywords%2Fnextjs"
          rel="nofollow noopener noreferrer"
          target="_blank"
          title="LINEでシェアする"
        >
          <svg
            class="w-7 sm:w-8 h-7 sm:h-8"
            fill="currentColor"
            height="24"
            stroke="none"
            viewBox="-0.5 -1 25 25"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.365 9.863c.349 0 .63.285.63.631 0 .345-.281.63-.63.63H17.61v1.125h1.755c.349 0 .63.283.63.63 0 .344-.281.629-.63.629h-2.386c-.345 0-.627-.285-.627-.629V8.108c0-.345.282-.63.63-.63h2.386c.346 0 .627.285.627.63 0 .349-.281.63-.63.63H17.61v1.125h1.755zm-3.855 3.016c0 .27-.174.51-.432.596-.064.021-.133.031-.199.031-.211 0-.391-.09-.51-.25l-2.443-3.317v2.94c0 .344-.279.629-.631.629-.346 0-.626-.285-.626-.629V8.108c0-.27.173-.51.43-.595.06-.023.136-.033.194-.033.195 0 .375.104.495.254l2.462 3.33V8.108c0-.345.282-.63.63-.63.345 0 .63.285.63.63v4.771zm-5.741 0c0 .344-.282.629-.631.629-.345 0-.627-.285-.627-.629V8.108c0-.345.282-.63.63-.63.346 0 .628.285.628.63v4.771zm-2.466.629H4.917c-.345 0-.63-.285-.63-.629V8.108c0-.345.285-.63.63-.63.348 0 .63.285.63.63v4.141h1.756c.348 0 .629.283.629.63 0 .344-.282.629-.629.629M24 10.314C24 4.943 18.615.572 12 .572S0 4.943 0 10.314c0 4.811 4.27 8.842 10.035 9.608.391.082.923.258 1.058.59.12.301.079.766.038 1.08l-.164 1.02c-.045.301-.24 1.186 1.049.645 1.291-.539 6.916-4.078 9.436-6.975C23.176 14.393 24 12.458 24 10.314"
            />
          </svg>
        </a>
        <a
          class="w-11 sm:w-14 h-11 sm:h-14 mx-2 sm:mx-4 rounded-full flex justify-center items-center hover:text-sns-hatena border border-shadow-light dark:border-shadow-dark hover:border-sns-hatena dark:hover:border-sns-hatena shadow-light hover:shadow-none dark:shadow-inset-dark dark:hover:shadow-none"
          href="http://b.hatena.ne.jp/add?&url=https%3A%2F%2Fblog.shinki.net%2Fkeywords%2Fnextjs"
          rel="nofollow noopener noreferrer"
          target="_blank"
          title="はてなブックマークでシェアする"
        >
          <svg
            class="w-5 sm:w-7 h-5 sm:h-7"
            fill="currentColor"
            height="24"
            stroke="none"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.47 0C22.42 0 24 1.58 24 3.53v16.94c0 1.95-1.58 3.53-3.53 3.53H3.53C1.58 24 0 22.42 0 20.47V3.53C0 1.58 1.58 0 3.53 0h16.94zm-3.705 14.47c-.78 0-1.41.63-1.41 1.41s.63 1.414 1.41 1.414 1.41-.645 1.41-1.425-.63-1.41-1.41-1.41zM8.61 17.247c1.2 0 2.056-.042 2.58-.12.526-.084.976-.222 1.32-.412.45-.232.78-.564 1.02-.99s.36-.915.36-1.48c0-.78-.21-1.403-.63-1.87-.42-.48-.99-.734-1.74-.794.66-.18 1.156-.45 1.456-.81.315-.344.465-.824.465-1.424 0-.48-.103-.885-.3-1.26-.21-.36-.493-.645-.883-.87-.345-.195-.735-.315-1.215-.405-.464-.074-1.29-.12-2.474-.12H5.654v10.486H8.61zm.736-4.185c.705 0 1.185.088 1.44.262.27.18.39.495.39.93 0 .405-.135.69-.42.855-.27.18-.765.254-1.44.254H8.31v-2.297h1.05zm8.656.706v-7.06h-2.46v7.06H18zM8.925 9.08c.71 0 1.185.08 1.432.24.245.16.367.435.367.83 0 .38-.13.646-.39.804-.265.154-.747.232-1.452.232h-.57V9.08h.615z"
            />
          </svg>
        </a>
        <a
          class="w-11 sm:w-14 h-11 sm:h-14 mx-2 sm:mx-4 rounded-full flex justify-center items-center hover:text-sns-pocket border border-shadow-light dark:border-shadow-dark hover:border-sns-pocket dark:hover:border-sns-pocket shadow-light hover:shadow-none dark:shadow-inset-dark dark:hover:shadow-none"
          href="https://getpocket.com/edit?url=https%3A%2F%2Fblog.shinki.net%2Fkeywords%2Fnextjs"
          rel="nofollow noopener noreferrer"
          target="_blank"
          title="Pocketでシェアする"
        >
          <svg
            class="w-6 sm:w-7 h-6 sm:h-7"
            fill="currentColor"
            height="24"
            stroke="none"
            viewBox="0 -1 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.813 10.259l-5.646 5.419c-.32.305-.73.458-1.141.458-.41 0-.821-.153-1.141-.458l-5.646-5.419c-.657-.628-.677-1.671-.049-2.326.63-.657 1.671-.679 2.325-.05l4.511 4.322 4.517-4.322c.66-.631 1.697-.607 2.326.049.631.645.615 1.695-.045 2.326l-.011.001zm5.083-7.546c-.299-.858-1.125-1.436-2.041-1.436H2.179c-.9 0-1.717.564-2.037 1.405-.094.25-.142.511-.142.774v7.245l.084 1.441c.348 3.277 2.047 6.142 4.682 8.139.045.036.094.07.143.105l.03.023c1.411 1.03 2.989 1.728 4.694 2.072.786.158 1.591.24 2.389.24.739 0 1.481-.067 2.209-.204.088-.029.176-.045.264-.06.023 0 .049-.015.074-.029 1.633-.36 3.148-1.036 4.508-2.025l.029-.031.135-.105c2.627-1.995 4.324-4.862 4.686-8.148L24 10.678V3.445c0-.251-.031-.5-.121-.742l.017.01z"
            />
          </svg>
        </a>
      </div>
    </aside>
    <div
      class="md:flex justify-around"
    >
      <aside
        class="mx-auto mt-10 sm:mt-16 lg:mt-20 table"
      >
        <h2
          class="mb-3 sm:mb-5 lg:mb-6 text-xl sm:text-2xl tracking-wider flex justify-center items-center"
        >
          <span
            class="text-accent-light dark:text-accent-dark text-2xl sm:text-3xl"
          >
            C
          </span>
          ONTENTS
        </h2>
        <ul
          class="flex"
        >
          <li
            class="mx-5 lg:mx-8 text-sm sm:text-base"
          >
            <a
              class="hover:text-accent-light dark:hover:text-accent-dark flex items-center"
              href="/keywords"
            >
              <svg
                class="w-6 h-6 mr-2"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6l3 1m0 0l-3 9a5.002 5.002 0 006.001 0M6 7l3 9M6 7l6-2m6 2l3-1m-3 1l-3 9a5.002 5.002 0 006.001 0M18 7l3 9m-3-9l-6-2m0-2v2m0 16V5m0 16H9m3 0h3"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
              キーワード一覧
            </a>
          </li>
        </ul>
      </aside>
      <aside
        class="mx-auto mt-10 sm:mt-16 lg:mt-20 table"
      >
        <h2
          class="mb-3 sm:mb-5 lg:mb-6 text-xl sm:text-2xl tracking-wider flex justify-center items-center"
        >
          <span
            class="text-accent-light dark:text-accent-dark text-2xl sm:text-3xl"
          >
            M
          </span>
          ORE
        </h2>
        <ul
          class="flex justify-center"
        >
          <li
            class="text-sm sm:text-base"
          >
            <a
              class="hover:text-accent-light dark:hover:text-accent-dark flex items-center"
              href="https://twitter.com/code_shinki"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              <svg
                class="w-6 h-6 mr-1"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z M15 11a3 3 0 11-6 0 3 3 0 016 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
              Twitter
            </a>
          </li>
          <li
            class="mx-4 sm:mx-6 lg:mx-9 text-sm sm:text-base"
          >
            <a
              class="hover:text-accent-light dark:hover:text-accent-dark flex items-center"
              href="https://github.com/Code-Shinki"
              rel="nofollow noopener noreferrer"
              target="_blank"
            >
              <svg
                class="w-6 h-6 mr-1"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z M15 11a3 3 0 11-6 0 3 3 0 016 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
              GitHub
            </a>
          </li>
          <li
            class="text-sm sm:text-base"
          >
            <div
              class="line-through flex items-center"
            >
              <svg
                class="w-6 h-6 mr-1"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z M15 11a3 3 0 11-6 0 3 3 0 016 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
              Shinki.net
            </div>
          </li>
        </ul>
      </aside>
    </div>
  </main>
  <footer
    class="pb-5 md:pb-8 lg:pb-10 mt-14 sm:mt-20 lg:mt-20 flex justify-center"
  >
    <small
      class="text-xs lg:text-sm italic"
    >
      (c) 2020 - 2021 Shinki all rights reserved.
    </small>
  </footer>
</DocumentFragment>
`;
